src/hardware/adlib.cpp

```diff
@@ -39,19 +40,24 @@
 #define OPL2_INTERNAL_FREQ    3600000   // The OPL2 operates at 3.6MHz
 #define OPL3_INTERNAL_FREQ    14400000  // The OPL3 operates at 14.4MHz
 
+static Adlib::Module *module = nullptr;
+static AdlibGold *adlib_gold = nullptr;
```

Comment:
Suggest managing these using `unique_ptr`:

``` c++
#include <memory>

...

static std::unique_ptr<Adlib::Module> module = {};
static std::unique_ptr<AdlibGold> adlib_gold = {};
```

Reply:
Can be done, but they're globals, so why does it matter? They still need to be managed manually on re-creation.

Reply:
> They still need to be managed manually on re-creation.

The smart pointer removes the need to differentiate between first-time creation and re-creation (as long as the object's destructor is healthy)

So repeated calls to:
``` c++
  adlib_gold = std::make_unique<AdlibGold>(mixerChan->GetSampleRate());
```

are fine (without needing any `if (adlib_gold) adlib_gold.reset();`)

Reply:
But I also see that you're following the "parent-level" design approach to adding this feature, so are using std::unique_ptr inside scope of the PR, but not pressing that opinionated design out further than the scope.  I'm all for that too, so marking this resolved.